sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -172,7 +172,7 @@ case class HiveTableScanExec(
         prunePartitions(hivePartitions)
       }
     } else {
-      if (sparkSession.sessionState.conf.metastorePartitionPruning &&
```

Comment:
let's keep this unchanged for now. We may override `def conf` in `SparkPlan` later, to always get conf from the captured spark session.

Reply:
Reverted